Xamarin.Forms.Platform.Android/CollectionView/RecyclerViewScrollListener.cs

```diff
@@ -0,0 +1,58 @@
+﻿using Android.Support.V7.Widget;
+
+namespace Xamarin.Forms.Platform.Android.CollectionView
+{
+	public class RecyclerViewScrollListener : RecyclerView.OnScrollListener
+	{
+		bool _disposed;
+		ItemsView _itemsView;
+		ItemsViewAdapter _itemsViewAdapter;
+
+		public RecyclerViewScrollListener(ItemsView itemsView, ItemsViewAdapter itemsViewAdapter)
+		{
+			_itemsView = itemsView;
+			_itemsViewAdapter = itemsViewAdapter;
+		}
+
+		public override void OnScrollStateChanged(RecyclerView recyclerView, int newState)
+		{
+			base.OnScrollStateChanged(recyclerView, newState);
+		}
+
+		public override void OnScrolled(RecyclerView recyclerView, int dx, int dy)
+		{
+			base.OnScrolled(recyclerView, dx, dy);
+
+			var linearLayoutManager = recyclerView.GetLayoutManager() as LinearLayoutManager;
```

Comment:
I assumed that every layout manager is a base `LinearLayoutManager`. Do you expect this to be false in the future?

Reply:
We expect that someone will eventually want to use a LayoutManager that is _not_ derived from LinearLayoutManager. So we try our best to do something reasonable when this happens. For now, I'd suggest just not firing the event if the LayoutManager is not a LinearLayoutManager.

Ideally, we'd add some sort of interface that other LayoutManagers could implement so we could interrogate them about the remaining items.

Reply:
When the layout manager is not LinearLayoutManager, then we send -1 for the item indices. This can be documented for devs.